From 49ecb703d9d0bfd38eb69ba5cb63a8262bd03f96 Mon Sep 17 00:00:00 2001
From: steve-chavez <stevechavezast@gmail.com>
Date: Thu, 11 Jun 2020 13:18:07 -0500
Subject: [PATCH] Add hasql-notifications openssl linking fix

---
 survey/default.nix | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/survey/default.nix b/survey/default.nix
index 828beaa..9c2d5f6 100644
--- a/survey/default.nix
+++ b/survey/default.nix
@@ -1054,6 +1054,11 @@ let
                   super.squeal-postgresql
                   [ final.openssl ]
                   "--libs openssl";
+              hasql-notifications =
+                addStaticLinkerFlagsWithPkgconfig
+                  super.hasql-notifications
+                  [ final.openssl ]
+                  "--libs openssl";

               xml-to-json =
                 addStaticLinkerFlagsWithPkgconfig
--
2.19.3

