diff --git a/survey/default.nix b/survey/default.nix
index 46d8066..a47f214 100644
--- a/survey/default.nix
+++ b/survey/default.nix
@@ -1519,7 +1519,7 @@ let
                   [
                     "--enable-executable-static" # requires `useFixedCabal`
                     # `enableShared` seems to be required to avoid `recompile with -fPIC` errors on some packages.
-                    "--extra-lib-dirs=${final.ncurses.override { enableStatic = true; enableShared = true; }}/lib"
+                    "--extra-lib-dirs=${final.ncurses}/lib"
                   ]
                   # TODO Figure out why this and the below libffi are necessary.
                   #      `working` and `workingStackageExecutables` don't seem to need that,
