From bb4c1e27e391eff01591fe60830ff68a9ada41ef Mon Sep 17 00:00:00 2001
From: Remo <remo>
Date: Wed, 22 Apr 2020 11:58:07 +0200
Subject: [PATCH] Add postgrest libpq linking fix

---
 survey/default.nix | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/survey/default.nix b/survey/default.nix
index b28606c..fd7cde7 100644
--- a/survey/default.nix
+++ b/survey/default.nix
@@ -1052,6 +1052,11 @@ let
                   super.squeal-postgresql
                   [ final.openssl ]
                   "--libs openssl";
+              postgrest =
+                addStaticLinkerFlagsWithPkgconfig
+                  super.postgrest
+                  [ final.openssl ]
+                  "--libs openssl";
 
               xml-to-json =
                 addStaticLinkerFlagsWithPkgconfig
